path_rotate_role.go

```diff
@@ -0,0 +1,74 @@
+package tfc
+
+import (
+	"context"
+
+	"github.com/hashicorp/vault/sdk/framework"
+	"github.com/hashicorp/vault/sdk/logical"
+)
+
+func pathRotateRole(b *tfBackend) []*framework.Path {
+	return []*framework.Path{
+		{
+			Pattern: "rotate-role/" + framework.GenericNameRegex("name"),
+			Fields: map[string]*framework.FieldSchema{
+				"name": {
+					Type:        framework.TypeString,
+					Description: "Name of the team or organization role",
+				},
+			},
+
+			Operations: map[logical.Operation]framework.OperationHandler{
+				logical.UpdateOperation: &framework.PathOperation{
+					Callback:                    b.pathRotateRole,
+					ForwardPerformanceStandby:   true,
+					ForwardPerformanceSecondary: true,
+				},
+			},
+
+			HelpSynopsis:    pathRotateRoleHelpSyn,
+			HelpDescription: pathRotateRoleHelpDesc,
+		},
+	}
+}
+
+func (b *tfBackend) pathRotateRole(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	name := d.Get("name").(string)
+	if name == "" {
+		return logical.ErrorResponse("missing role name"), nil
+	}
+
+	roleEntry, err := b.getRole(ctx, req.Storage, name)
+	if err != nil {
+		return nil, err
+	}
+
+	if roleEntry == nil {
+		return logical.ErrorResponse("missing role entry"), nil
+	}
+
+	if roleEntry.UserID != "" {
+		return logical.ErrorResponse("cannot rotate credentials for user roles"), nil
+	}
+
+	token, err := b.createToken(ctx, req.Storage, roleEntry)
+	if err != nil {
+		return nil, err
+	}
+
+	roleEntry.Token = token.Token
+
+	if err := setTerraformRole(ctx, req.Storage, name, roleEntry); err != nil {
```

Comment:
Nit: Can we rename the func to `setRole` so that it matches the getter, `getRole`?

Reply:
Yup, done in 0da18ba